explorer/src/hooks/useConsensusData.ts

```diff
@@ -13,21 +13,23 @@ import { formatDeposits, formatOperators, formatWithdrawals } from 'utils/chainS
 export const useConsensusData = () => {
   const [isLoading, setIsLoading] = useState(false)
   const [isLoaded, setIsLoaded] = useState(false)
-  const {
-    setProperties,
-    setDomain,
-    setSystem,
-    setDomainRegistry,
-    setDomainStakingSummary,
-    setLatestConfirmedDomainExecutionReceipt,
-    setNominatorCount,
-    setOperatorIdOwner,
-    setOperators,
-    setPendingStakingOperationCount,
-    setSuccessfulBundles,
-    setDeposits,
-    setWithdrawals,
-  } = useConsensusStates()
+  const setProperties = useConsensusStates((state) => state.setProperties)
+  const setDomain = useConsensusStates((state) => state.setDomain)
+  const setSystem = useConsensusStates((state) => state.setSystem)
+  const setDomainRegistry = useConsensusStates((state) => state.setDomainRegistry)
+  const setDomainStakingSummary = useConsensusStates((state) => state.setDomainStakingSummary)
+  const setLatestConfirmedDomainExecutionReceipt = useConsensusStates(
+    (state) => state.setLatestConfirmedDomainExecutionReceipt,
+  )
+  const setNominatorCount = useConsensusStates((state) => state.setNominatorCount)
+  const setOperatorIdOwner = useConsensusStates((state) => state.setOperatorIdOwner)
+  const setOperators = useConsensusStates((state) => state.setOperators)
+  const setPendingStakingOperationCount = useConsensusStates(
+    (state) => state.setPendingStakingOperationCount,
+  )
+  const setSuccessfulBundles = useConsensusStates((state) => state.setSuccessfulBundles)
+  const setDeposits = useConsensusStates((state) => state.setDeposits)
+  const setWithdrawals = useConsensusStates((state) => state.setWithdrawals)
```

Comment:
Why not use destructuring?

Reply:
By destructuring, we are fetching everything from the store leading to re-rendering of the component on any states changes, even states we don't use
https://github.com/pmndrs/zustand?tab=readme-ov-file#fetching-everything

Reply:
interesting, thinks I didn't know. this repeated code is so ugly though 😆.

Reply:
It could be made better by creating set function inside the store that set the states for all the values at the same time, but realistically I don't believe I use all of them, so I'll trim a lot of these soon, ideally the indexer should have all of these and we should query from indexer